FRUI/FRUI/ViewControllers/AuthStepViewController.swift

```diff
@@ -126,7 +126,33 @@ class AuthStepViewController: UIViewController {
                 //  Set auth callbacks in list for rendering
                 self.currentNode = node
                 self.authCallbacks = node.callbacks
-                self.renderAuthStep()
+                
+                //  If DeviceProfileCallback is the only Callback, then immediately collect, and submit
+                if self.authCallbacks.count == 1, let deviceProfileCallback = self.authCallbacks.first as? DeviceProfileCallback {
+                    deviceProfileCallback.execute { (profile) in
+                        self.submitNode()
+                    }
+                }
+                else {
+                    var deviceProfileCallback: DeviceProfileCallback?
+                    for callback in self.authCallbacks {
+                        if let thisCallback = callback as? DeviceProfileCallback {
+                            deviceProfileCallback = thisCallback
+                        }
+                    }
+                    //  If DeviceProfileCallback is found as one of Callbacks, collect data first
+                    if let deviceProfileCallback = deviceProfileCallback {
+                        self.startLoading()
```

Comment:
May I ask why you're blocking the `renderAuthStep` until after you have collected the device profile?

Reply:
@cerebrl , renderAuthStep basically renders all Callback in the Node in the list view. If the DeviceProfileCallback is found as one of Callbacks in the list, then from line #144 - #150, what it does is to collect DeviceProfile information first before render all Callback, and upon completion, it renders the UI.

The reason for that is, as DeviceProfile collecting operation is async and the submit action can basically happen at anytime after the UI is rendered, I wanted to make sure that the device information is collected before submitting the node by the user.

Reply:
Thanks for the explanation, @jhoongo. You bring up a good point about the possible race condition. Are you rendering any kind of spinner or "waiting ... " message since the user may be presented with a blank screen?

Reply:
From my testing, no spinner is displayed... but this is expected per he acceptance criteria of SDKS-436:
"Don't display "loading indicator" / "spinner""
"Don't display message"

However, as we all agree, it is much safer to wait for the data collection to finish and then render the form...

Reply:
@spetrov , I've updated the code to better visualize loading indicator when there is only DeviceProfileCallback. Instead of having a generic loading indicator that goes away quickly, I re-render the TableView with dedicated loading view cell for DeviceProfileCallback. Please let me know if this makes better sense. Thanks!

Reply:
Yes, it's much better now! Thanks!